checks.d/haproxy.py

```diff
@@ -319,51 +336,37 @@ def _process_status_metric(self, hosts_statuses, collect_status_metrics_by_host,
                 service, status = host_status
             status = status.lower()
 
-            tags = []
-            if count_status_by_service:
-                tags.append('service:%s' % service)
-
             if self._is_service_excl_filtered(service, services_incl_filter, services_excl_filter):
                 continue
 
+            tags = []
+            if count_status_by_service:
+                tags.append('service:%s' % service)
             if collect_status_metrics_by_host:
                 tags.append('backend:%s' % hostname)
```

Comment:
good point, `hostname` is supposed to be there since it's extracted [here](https://github.com/DataDog/dd-agent/blob/5.8.2/checks.d/haproxy.py#L237-L240), but the extraction could potentially not work as expected if the collected data isn't valid.

It's kind of outside the scope of this PR but it should be handled in a better way, agreed, I'll work on that